src/Language/Fixpoint/Solver/Common.hs

```diff
@@ -16,7 +16,7 @@ askSMT :: Config -> Context -> [(Symbol, Sort)] -> Expr -> IO Bool
 askSMT cfg ctx xs e
 --   | isContraPred e  = return False
   | isTautoPred  e     = return True
-  | null (kvarsExpr e) = checkValidWithContext ctx [] PTrue e'
+  | null (kvarsExpr e) = checkValidWithContext ctx xs PTrue e'
```

Comment:
I think it was impossible to trigger a crash before my change, xs is a list containing all the variables in scope, so they are actually needed when checking that e' is valid as e' it can reference some variables in the outer scope

Reply:
> so they are actually needed when checking that e' is valid as e'

These might have been already given to the smt solver before calling `askSMT`, or they might not be needed. Otherwise, tests would be failing because of undeclared identifiers.

I think there is still not enough of a motivation for this change.

Reply:
If I remember correctly (and from the calls of this function) this `xs` indeed is meant to give to the SMT the lambda arguments. Since up to now LH does not really have support for lambdas, it makes sense that this was overlooked. 

If @AlecsFerra needs this edit to make this code work and it does not make any of the existing tests failing, I suggest we leave it.

Reply:
If you try to run this https://gist.github.com/AlecsFerra/c31162353b21e034dc3797c5c227c049 without passing xs its going to crash with error `  crash: SMTLIB2 respSat = Error "line 28 column 3941: unknown constant eta$35$$35$0"`

Reply:
That looks good enough. Thanks.